src/services/rocksdb/backend.rs

```diff
@@ -0,0 +1,253 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::fmt::Formatter;
+use std::io::Error;
+use std::io::ErrorKind;
+use std::io::Result;
+use std::path::Path;
+use std::path::PathBuf;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::ready;
+use std::task::Context;
+use std::task::Poll;
+use std::vec::IntoIter;
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use futures::future::BoxFuture;
+use futures::Future;
+use futures::Stream;
+use pin_project::pin_project;
+use rocksdb::TransactionDB;
+
+use crate::adapters::kv;
+use crate::error::new_other_backend_error;
+use crate::Accessor;
+use crate::AccessorCapability;
+use crate::Scheme;
+
+const SCAN_LIMIT: usize = 100;
+
+/// Rocksdb backend builder
+#[derive(Clone, Default, Debug)]
+pub struct Builder {
+    /// The path to the rocksdb database
+    path: Option<PathBuf>,
```

Comment:
I prefer to use `String` to make our API cleaner. Nearly no user in OpenDAL will need to use non utf-8 pathes.

Reply:
Make sense